stock_by_warehouse/static/src/components/warehouse/warehouse_field.xml

```diff
@@ -27,61 +27,84 @@
+                                line.available gt 0
+                                or line.outgoing gt 0
+                                or line.saleable gt 0
+                                or line.incoming gt 0
+                                or line.virtual gt 0
                             "
-                            class="list-group-item row"
                         >
-                            <div name="warehouse_widget" class="d-flex justify-content-between">
-                                <b class="text-left" t-att-title="'Warehouse Name: ' + line.warehouse">
-                                    <t t-out="line.warehouse_short"> | </t>
-                                </b>
-                                <div class="text-right">
-                                    <span
-                                        title="This is real(on hand) quantity  minus reserved quantity"
-                                        class="text-muted"
-                                        t-out="line.available_not_res_formatted"
+                            <tr>
+                                <td style="text-align: center;">
+                                    <b
+                                        class="text-left"
+                                        t-att-title="'Warehouse Name: ' + line.warehouse"
+                                        t-out="line.warehouse_short"
                                     />
-                                    <br />
+                                </td>
+                                <td style="text-align: center;">
                                     <span
-                                        title="Available now but not necessarily what I can sale"
+                                        class="text-muted"
+                                        title="Available now, but not necessarily for sale"
                                         t-out="line.available_formatted"
                                     />
-                                    <br />
-                                    - <span title="Outgoing" class="text-danger" t-out="line.outgoing_formatted" />
-                                    <br />
-                                    = <span
-                                        title="I can sale this quantity now and deliver it immediately"
-                                        class="text-muted"
+                                </td>
+                                <td style="text-align: center;">
+                                    <span class="text-muted" title="Outgoing" t-out="line.outgoing_formatted" />
+                                </td>
+                                <td style="text-align: center;">
+                                    <t t-if="line.virtual gt 0" t-set="saleable_qty_title_message">
+                                        <t> This quantity is available for immediate sale and delivery.</t>
```

Comment:
Sorry for not noticing it before:

```suggestion
                                        <t>This quantity is available for immediate sale and delivery.</t>
```

Same below.

Reply:
No worries! No need to apologize 😉. Done!